symengine/series_generic_multivariate.cpp

```diff
@@ -0,0 +1,574 @@
+                                      const map_basic_uint precs)
+{
+    // If the series only has a constant term, then degree = 0, var is irrelvant
+    if (p.vars_.empty()) {
+        /*  if (degree != 0)
+              //return false;
+          if (var != "")
+              //return false;*/
+    } else {
+        // check that p is a polynomial in terms of symbol(var)
+        if (p.vars_.find(symbol(var)) == p.vars_.end()) {
+            return false;
+        }
+
+        // check that degree is the precision of "var" in precs.
+        if (precs.find(symbol(var)) != precs.end()) {
+            if (precs.find(symbol(var))->second
+                != static_cast<unsigned int>(degree))
+                return false;
+        } else {
+            return false;
+        }
+    }
+
+    // check that the precision in precs define precisions for all of the
+    // symbols of p and that these precisions are less than the degree of those
+    // variables in p
+    for (RCP<const Basic> s : p.vars_) {
+        if (precs.find(s) == precs.end())
+            return false;
+        if (p.degrees_.find(s)->second > 0
+            && precs.find(s)->second
+                   < static_cast<unsigned int>(p.degrees_.find(s)->second))
+            return false;
+    }
+
+    return true;
+}
+
+RCP<const MultivariateSeries>
+MultivariateSeries::series(const RCP<const Basic> &t, const std::string &x,
+                           unsigned int prec)
+{
+    MultivariatePolynomialExpr p
+        = MultivariatePolynomialExpr::create({symbol(x)}, {{{1}, 1}});
+    SeriesVisitor<MultivariatePolynomialExpr, Expression, MultivariateSeries>
+        visitor(std::move(p), x, prec);
+    return visitor.series(t);
+}
+
+std::size_t MultivariateSeries::__hash__() const
+{
+    std::size_t seed = MULTIVARIATESERIES;
+    seed ^= p_.__hash__() + 0x9e3779b + (seed << 6) + (seed >> 2);
+    for (auto bucket : precs_) {
+        seed ^= bucket.second + 0x9e3779b + (seed << 6) + (seed >> 2);
+    }
+
+    return seed;
+}
+
+int MultivariateSeries::compare(const Basic &other) const
+{
+    SYMENGINE_ASSERT(is_a<MultivariateSeries>(other))
+    const MultivariateSeries &o
+        = static_cast<const MultivariateSeries &>(other);
+    int cmp = p_.compare(o.p_);
+    if (cmp != 0)
+        return cmp;
+
+    return map_basic_uint_compare(precs_, o.precs_);
+}
+
+RCP<const Basic> MultivariateSeries::as_basic() const
+{
+    return p_.get_basic();
+}
+
+umap_int_basic MultivariateSeries::as_dict() const
+{
+    umap_int_basic map;
+    // for (int i = 0; i <= get_degree(); i++)
+    // map[i] = p_.get_basic()->dict_.at(i).get_basic();
+    return map;
+}
+
+bool MultivariateSeries::__eq__(const Basic &o) const
+{
+    // ignore var_ and degree_, since thier information should be contained in
+    // p_.vars_ and precs_
+    return (is_a<MultivariateSeries>(o)
+            and p_ == static_cast<const MultivariateSeries &>(o).p_
+            and map_basic_uint_eq(
```

Comment:
Just tried it out, it works. `==` is defined for both the keys and values of the map. @myluszczak  You can remove `map_basic_uint_eq`